File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    result = 1
    
    while n != 1:
        if n % 2 == 0:
            n = n // 2
        else:
            result += 1
            n -= 1
            
    return result

'''
k칸 점프 (k만큼 건전지 사용) / 현재거리 x 2 순간이동 (건전지 사용 안함)
'''

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.js
const solution = (n) => {
    let result = 1
    
    while (n != 1) {
        if (n % 2 === 0) {
            n = Math.floor(n / 2);
        } else {
            result += 1
            n -= 1
        }
    }
    return result;
}